src/GovUk.Education.ExploreEducationStatistics.Data.Services/ResultSubjectMetaService.cs

```diff
@@ -281,58 +281,39 @@ private static string OrderLocationAttributes(LocationAttributeNode node)
             {
-                if (!GeoJsonRequested)
+                var selectedBoundaryLevel = _boundaryLevelRepository
+                    .FindOrNotFound(boundaryLevelId)
+                    .OrElse(() => null!)
+                    .Right;
+
+                if (selectedBoundaryLevel == null || !locations.ContainsKey(selectedBoundaryLevel.Level))
                 {
                     return new Dictionary<GeographicLevel, Dictionary<string, GeoJson>>();
                 }
-
+                
                 return locations.ToDictionary(
                     pair => pair.Key,
                     pair =>
                     {
                         var (geographicLevel, locationAttributes) = pair;
-                        return GetGeoJson(
-                            geographicLevel,
-                            locationAttributes
-                                .SelectMany(node => node.GetLeafAttributes())
-                                .ToList());
-                    });
-            }
-
-            private Dictionary<string, GeoJson> GetGeoJson(
-                GeographicLevel geographicLevel,
-                IReadOnlyList<ILocationAttribute> locationAttributes)
-            {
-                if (!GeoJsonRequested)
-                {
-                    return new Dictionary<string, GeoJson>();
-                }
-
-                // If no boundary level is requested, get the latest boundary level id for the geographic level
-                var boundaryLevelId = _query.BoundaryLevel ?? GetLatestBoundaryLevelByGeographicLevel(geographicLevel);
 
-                // Not all geographic levels have boundary level data configured so expect to return an empty result
-                if (boundaryLevelId == null)
-                {
-                    return new Dictionary<string, GeoJson>();
-                }
-
-                var locationAttributeCodes = locationAttributes
-                    .Select(locationAttribute => locationAttribute.GetCodeOrFallback())
-                    .WhereNotNull();
-
-                return _geoJsonRepository.FindByBoundaryLevelAndCodes(boundaryLevelId.Value, locationAttributeCodes);
+                        if (geographicLevel != selectedBoundaryLevel.Level)
+                        {
+                            return new Dictionary<string, GeoJson>();
+                        }
+                        
+                        var locationAttributeCodes = locationAttributes
+                            .SelectMany(locationAttribute => locationAttribute.GetLeafAttributes())
+                            .Select(locationAttribute => locationAttribute.GetCodeOrFallback())
+                            .WhereNotNull();
+
+                        return _geoJsonRepository.FindByBoundaryLevelAndCodes(boundaryLevelId, locationAttributeCodes);
+                    });
             }
-
-            private bool GeoJsonRequested => _query.IncludeGeoJson != null && _query.IncludeGeoJson.Value;
```

Comment:
Previously `IncludeGeoJson` on the query was set to true in `GetReleaseDataBlockResults` if the data block contained a map, and here we see `GeoJsonRequested ` was true when `IncludeGeoJson`, and that in turn controlled whether GeoJson was returned or not.

It now seems like we are returning GeoJson all of the time meaning that we could be returning it for table builder results where the geojson is not required  because no chart is rendered in the table builder.

EES-3332 relates to this and suggests this isn't new behaviour but I think it's worth checking again as I think an adverse change could be being made here. We wouldn't want the geojson being returned in table builder responses.

Reply:
Yup, checked it out on the dev branch and we are indeed currently returning GeoJSON when viewing a data block table, if that data block has a Map configured.